run/run.go

```diff
@@ -111,7 +115,7 @@ func GetMain() *cobra.Command {
 
 	// help and documentation
 	cmd.InitDefaultHelpCmd()
-	cmd.AddCommand(kptcommands.GetKptCommands("kpt")...)
+	cmd.AddCommand(kptcommands.GetKptCommands("kpt", newFactory(cmd))...)
```

Comment:
Should we use the newFactory to refresh OpenAPI and discovery in the background immediately?